### PR TITLE
decorate metrics with haproxyClusterName

### LINKED ISSUE #28: `clusterName` metric attribute collides with Kubernetes'
[NOTE]: # ( ^^ Provide a general summary of the issue in the title above. ^^ )

## Description
[NOTE]: # ( Describe the problem you're encountering. )
[TIP]:  # ( Do NOT share sensitive information, whether personal, proprietary, or otherwise! )

HAProxy integration writes the name of the HAProxy cluster as a `clusterName` attribute in metrics. However, this label is also used by the Kubernetes integrations to annotate the Cluster name in which an integration is running. This causes the latter to shadow the former.

## Expected Behavior
[NOTE]: # ( Tell us what you expected to happen. )

HAproxy ClusterName and Kubernetes ClusterName should be reported as different attributes.

## Steps to Reproduce
[NOTE]: # ( Please be as specific as possible. )
[TIP]:  # ( Link a sample application that demonstrates the issue. )

Run the HAProxy integration in Kubernetes.

### PR DESCRIPTION
**Pull request by roobre**
`clusterName` attribute might get overwritten by the kubernetes integration when running inside kubernetes.
This hotfix allows to identify different haproxy clusters that are located in the same Kubernetes cluster.

Fixes #28

### CODE REVIEW

**Review by paologallinaharbur — APPROVED** [review]

File: src/collection.go [hunk 1]
```diff
@@ -156,6 +156,8 @@ func collectMetricsOfType(entityType string, definitions map[string]metricDefini
 	}
 
 	ms := e.NewMetricSet(fmt.Sprintf("HAProxy%sSample", strings.Title(strings.TrimPrefix(entityType, "ha-"))),
+		// Decorate with haproxyClusterName as well, since clusterName might be overwritten by nri-kubernetes
+		attribute.Attribute{Key: "haproxyClusterName", Value: args.ClusterName},
```
Comment (arvdias) [inline src/collection.go:160]: I'm not 100% sure if  you need to add this to the integration spec files..

**Review by arvdias — APPROVED** [review]

**Comment by arvdias** [comment]
Does just adding the new attribute change anything? Shouldn't we also include the clusterName as part of the entity Id? This [here](https://github.com/newrelic/nri-haproxy/blob/040f9426a8b5c6b7f5f00f428db4f713a4cf0771/src/collection.go#L178) does not seem to be to be sufficient to have an entity identifier that is unique in case we have multiple haproxy clusters running and being monitored (like in the K8s case) right?

**Comment by gsanchezgavier** [comment]
these changes will allow to keep using the cluster_name (for backward compatibility reasons) but introduces a new parameter to avoid the collision with the k8s integration.
HA_PROXY_CLUSTER_NAME take precedence over CLUSTER_NAME.
Entity uniqueness will be done selecting different names for each HAProxy cluster. In k8s this con be done using the discover.name variable as is stated in the config example for k8s.

**Comment by gsanchezgavier** [comment]
go [here](https://staging.one.nr/0M8jqx6vRlv) to have a quick look of an example

**Comment by gsanchezgavier** [comment]
something worth to mention is that the `CLUSTER_NAME` env variable is configured as an `NRIA_PASSTHROUGH_ENVIRONMENT` in nri-kubernetes so even if it is added in the config the agent is using the one from k8s , that why we need to add a different parameter

**Review by arvdias — APPROVED** [review]

**Pull request merged**